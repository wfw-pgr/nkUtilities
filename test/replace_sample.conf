# this is a sample of variable definition
# -- below is definitions. -- #

# <define>   @coef1 = 1.0   # this is coef1
# <define>   @coef2 = 1000
# <define>   @coef3 = @coef1 + @coef2

$ <define>   @coef1 = 3.0   $ this is coef1
$ <define>   @coef2 = 22
$ <define>   @coef3 = @coef1 + @coef2
$ <define>   @type  = type-A

* <define>   @coef1 = 3.0   * this is coef1
* <define>   @coef2 = 22
* <define>   @coef3 = @coef1 + @coef2

# -- to disable the definition, use ## or ### etc. -- #
# -- program catch following expressions "#\s*@(variable's name)\s# -- #

set01	     0.1   0.2       @coef3
set02	     0.2   @coef1    @coef2
Set03	     0.1   0.2       0.4
Set04	     0.1   0.2       @coef2
type	     @type

$ <include> filepath = test/include.conf
