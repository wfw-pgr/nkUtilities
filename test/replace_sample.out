# this is a sample of variable definition
# -- below is definitions. -- #

# <define>   3.0 = 1.0   # this is coef1
# <define>   22 = 1000
# <define>   25.0 = 3.0 + 22

$ <define>   @coef1 = 3.0   $ this is coef1
$ <define>   @coef2 = 22
$ <define>   @coef3 = @coef1 + @coef2
$ <define>   @type  = type-A


# -- to disable the definition, use ## or ### etc. -- #
# -- program catch following expressions "#\s*@(variable's name)\s# -- #

set01	     0.1   0.2       25.0
set02	     0.2   3.0    22
Set03	     0.1   0.2       0.4
Set04	     0.1   0.2       22
type	     type-A
val_frm_tbl  value_from_table

$ <include> filepath = test/include.conf

$ <postProcess> echo "This is an echo test."
$ <postProcess> echo "This is an echo test."
$ <postProcess> cp test/replace_sample.conf test/replace_copy.conf